controllers/machinedeletionremediation_controller_test.go

```diff
@@ -165,19 +171,36 @@ var _ = Describe("Machine Deletion Remediation CR", func() {
 					verifyMachineIsDeleted(workerNodeMachineName)
 					verifyMachineNotDeleted(masterNodeMachineName)
 
-					// The transition of the Processing Condition status from
-					// unset to "Started", and finally to "Finished" is too
-					// fast to test the initial value ("Started") by inspecting
-					// the actual MDR CR. For this reason the initial value is not
-					// tested here.
+					// Machine is deleted, but the remediation is not completed yet
+					verifyConditionsMatch([]expectedCondition{
+						{commonconditions.ProcessingType, metav1.ConditionTrue, remediationStarted},
+						{commonconditions.SucceededType, metav1.ConditionUnknown, remediationStarted},
+						// Cluster provider is not set in this test
+						{commonconditions.PermanentNodeDeletionExpectedType, metav1.ConditionUnknown, v1alpha1.MachineDeletionOnUndefinedProviderReason}})
+
+					// Mock Machine and Node re-provisioning (even though this test does not actually delete the node, just the machine).
+					// 1. Create a Machine's replacement with a new name
+					// 2. Update WorkerNode's annotation to point to the new Machine
+					machineReplacementName := workerNodeMachineName + "-replacement"
+					workerNode.Annotations[machineAnnotationOpenshift] = fmt.Sprintf("%s/%s", defaultNamespace, machineReplacementName)
+					workerNodeMachineReplacement := createWorkerMachine(machineReplacementName)
+					Expect(k8sClient.Update(context.Background(), workerNode)).To(Succeed())
+					DeferCleanup(k8sClient.Delete, workerNodeMachineReplacement)
+					Expect(k8sClient.Create(context.Background(), workerNodeMachineReplacement)).To(Succeed())
+
+					// Wait for the Machine replacement to be available in the cluster
+					//Eventually(func() bool {
+					//m := &v1beta1.Machine{}
+					//k8sClient.Get(context.Background(), client.ObjectKeyFromObject(workerNodeMachineReplacement), m)
+					//return m != nil
+					//}, "60s").Should(BeTrue())
```

Comment:
If this check is discarded, then can it be deleted or do you plan to revisit this check later?

Reply:
oh, no, this is debug code I forgot to remove, thanks!

Reply:
>It's needed

Then we should we not comment these lines :)

Reply:
well, it's needed _to avoid asking 60s timeout to `verifyConditionMatches`_ :D 
which it is doing in this version.
However, since this 60s are needed only in this specific case, it is better to wait the machine and use `verifyConditionMatches` "normally" with 20s timeout.